File: src/main/java/com/condominium/management/unit/entity/Unit.java
package com.condominium.management.unit.entity;

import com.condominium.management.block.entity.Block;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "units")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Unit {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String number;

    private Integer floor;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private UnitType type;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private UnitStatus status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "block_id", nullable = false)
    private Block block;

    @Column(nullable = false)
    private LocalDateTime createdAt;
}

File: src/types/unit.ts
export interface Unit {
    id: number;
    number: string;
    floor: number | null;
    type: string;
    status: string;
    blockId: number;
    blockName: string;
}

export interface UnitRequest {
    number: string;
    floor: number | null;
    type: string;
    status: string;
    blockId: number;
}